lapis2/src/main/kotlin/org/genspectrum/lapis/silo/SiloClient.kt

```diff
@@ -0,0 +1,47 @@
+package org.genspectrum.lapis.silo
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse.BodyHandlers
+
+class SiloClient(private val siloUrl: String) {
+    private val objectMapper = jacksonObjectMapper()
+
+    fun <ResponseType> sendQuery(query: SiloQuery<ResponseType>): ResponseType {
+        val client = HttpClient.newHttpClient()
+        val request = HttpRequest.newBuilder(URI("$siloUrl/query"))
+            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+            .POST(HttpRequest.BodyPublishers.ofString(objectMapper.writeValueAsString(query)))
+            .build()
+
+        val response =
+            try {
+                client.send(request, BodyHandlers.ofString())
+            } catch (exception: Exception) {
+                throw SiloException(exception.message, exception)
+            }
+
+        if (response.statusCode() >= 400) {
```

Comment:
Why not check `!= 200`?

Reply:
Good question. I'll change it to `!= 200` ;)